### PR TITLE
Don't return an error from re1.Match.

### PR DESCRIPTION
**Pull request by eaburns**
The interface is prettier without the error.
Instead of the original apporach of panicing on IO errors,
store errors in the Runes implementer and check it
before determining if a match succeeds.

### CODE REVIEW

**Comment by eaburns** [comment]
Thoughts?

**Comment by eaburns** [comment]
Now, the implementer of re1.Runes needs to track its own errors. But that seems appropriate anyway, right?

**Comment by mccoyst** [comment]
OK, but none of the tests test the case of an error. They still have `if err != nil`, but err is no longer reassigned.

**Comment by eaburns** [comment]
Ew. I'll fix it and get back to you.

Also, do you only think it's OK because the test blunder or do you not love it in general?

**Comment by mccoyst** [comment]
I feel pretty weird about it, but it seems better than the error. I was looking to the standard regexp package, and its MatchReader doesn't return an error, even though RuneReader.ReadRune can return one. So there's precedent, and that case goes even further and doesn't do anything special with storing errors like you have here.

**Comment by eaburns** [comment]
I only see one instance of `err == nil` that was not reassigned. It was in re1_test.go in `regexpTest.run`. Did you find any others?

**Comment by eaburns** [comment]
Actually, hold off. I want to add some tests failing IO.

File: runes/runes.go [hunk 1]
```diff
@@ -38,6 +38,12 @@ type Buffer struct {
 	size int64
 }
 
+// A ReaderWriterAt implements the io.ReaderAt and io.WriterAt interfaces.
+type ReaderWriterAt interface {
```
Comment (eaburns) [inline runes/runes.go]: Is this defined in the standard library anywhere?

**Comment by eaburns** [comment]
PTAL.

**Comment by mccoyst** [comment]
TestReuse has two `err != nil` checks that I was thinking of.

**Comment by eaburns** [comment]
PTAL.

**Pull request merged**